ext/standard/array.c

```diff
@@ -349,6 +350,22 @@ static zend_always_inline int php_array_data_compare_unstable_i(Bucket *f, Bucke
 }
 /* }}} */
 
+static zend_always_inline int php_array_data_compare_strict_unstable_i(Bucket *f, Bucket *s) /* {{{ */
+{
+	if (zend_is_identical(&f->val, &s->val)) {
+		return 0;
+	}
+	int c = zend_compare(&f->val, &s->val);
+	if (c != 0) {
+		return c;
+	}
+	// types must be different; compare based on type.
```

Comment:
Arrays and objects have different strict/weak comparison semantics. You can have two arrays/objects that are `!==` and `==`, and you need to specify a well-defined ordering for these.

Reply:
For arrays these are two arrays which have the exact same contents.  These have the same semantics for `==` and `===`:
```
1] boris> $a = [1,2,3];
 → array(
  0 => 1,
  1 => 2,
  2 => 3
)
[2] boris> $b = [1,2,3];
 → array(
  0 => 1,
  1 => 2,
  2 => 3
)
[3] boris> $a == $b;
 → true
[4] boris> $a === $b;
 → true
```
So that shouldn't be a problem.  The object case /is/ an issue:
```
[6] boris> $a = (object)$a;
 → object(stdClass)(
  0 => 1,
  1 => 2,
  2 => 3
)
[7] boris> $b = (object)$b;
 → object(stdClass)(
  0 => 1,
  1 => 2,
  2 => 3
)
[8] boris> $a == $b;
 → true
[9] boris> $a === $b;
 → false
```
I propose to compare these based on `spl_object_id`. That should work, right?